src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -246,11 +246,14 @@ function VisEditor($scope, $route, timefilter, AppState, $window, kbnUrl, courie
 
     // update the searchSource when filters update
     $scope.$listen(queryFilter, 'update', function () {
+      savedVis.searchSource.set('filter', $scope.state.filters);
```

Comment:
When you have a filter applied in the visualization app, and press refresh, that filter is not in effect until you disable and reenable it. I think it's caused by us setting the filter to the searchSource only in case the queryFilter updates, which most likely won't be true when refreshing. So I think we should actually set this kind of the same way as query, or set it at least initially.